File: February 21, 2026.js
function scoreCurling(house) {

  const distR = [];
  const distY = [];

  function getRing(i, j) {
    if (i === 2 && j === 2) {    
      return 0;
    }

    if (i >= 1 && i <= 3 && j >= 1 && j <= 3) {
      return 1;
    }

    return 2;
  }

  for (let i = 0; i < 5; i++) {
    for (let j = 0; j < 5; j++) {

      if (house[i][j] === "R") {
        distR.push(getRing(i, j));
      }

      if (house[i][j] === "Y") {
        distY.push(getRing(i, j));
      }

    }
  }

  if (distR.length === 0 && distY.length === 0) {
    return "No points awarded";
  }

  distR.sort((a, b) => a - b);
  distY.sort((a, b) => a - b);

  const closestR = distR.length ? distR[0] : Infinity;
  const closestY = distY.length ? distY[0] : Infinity;

  if (closestR === closestY) {
    return "No points awarded";
  }

  if (closestR < closestY) {
    let points = 0;

    for (let d of distR) {
      if (d < closestY) points++;
      else break;
    }

    return `R: ${points}`;
  }

  if (closestY < closestR) {
    let points = 0;

    for (let d of distY) {
      if (d < closestR) points++;
      else break;
    }

    return `Y: ${points}`;
  }

  return "No points awarded";
}

File: February 21, 2026.py
def score_curling(house):

    dist_r = []
    dist_y = []

    def get_ring(i, j):
        if i == 2 and j == 2:
            return 0

        if 1 <= i <= 3 and 1 <= j <= 3:
            return 1

        return 2

    for i in range(5):
        for j in range(5):

            if house[i][j] == "R":
                dist_r.append(get_ring(i, j))

            if house[i][j] == "Y":
                dist_y.append(get_ring(i, j))

    if len(dist_r) == 0 and len(dist_y) == 0:
        return "No points awarded"

    dist_r.sort()
    dist_y.sort()

    closest_r = dist_r[0] if dist_r else float("inf")
    closest_y = dist_y[0] if dist_y else float("inf")

    if closest_r == closest_y:
        return "No points awarded"

    if closest_r < closest_y:
        points = 0
        for d in dist_r:
            if d < closest_y:
                points += 1
            else:
                break
        return f"R: {points}"

    if closest_y < closest_r:
        points = 0
        for d in dist_y:
            if d < closest_r:
                points += 1
            else:
                break
        return f"Y: {points}"

    return "No points awarded"